docs/query-builder.md

```diff
@@ -448,64 +448,161 @@ DB::collection('items')
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots');
+  ->where('name', 'John')
+  // Push a single value to the items array
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  // Push multiple values to the items array
+  ->push('items', ['hat', 'jeans']);
+// Result:
+// items: ['boots', 'hat', 'jeans']
 
+// Or
+
+// Push the value to an exact document
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add an array as a value to the messages array. value arrays must be nested into a list array, otherwise they are
+interpreted as a list of scalar values.
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an array as a value to the items array
+  ->push('messages', [
+      [
+          'from'    => 'Jane Doe',
+          'message' => 'Hi John',
+      ],
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
 
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots', true);
+  ->where('name', 'John')
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  ->push('items', ['hat', 'boots', 'jeans'], true);
+// Result:
+// items: ['boots', 'hat', 'jeans']
+
+// Or
+
+$user->push('messages', [
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
 
-$user->push('items', 'boots', true);
+$user->push('messages', [
+    [
+        'from'    => 'Jess Doe',
+        'message' => 'Hi',
+    ],
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
+], true);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
 ```
 
 **Pull**
 
-Remove an item from an array.
+Remove one or multiple values form items array.
 
 ```php
+// items: ['boots', 'hat', 'jeans']
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('items', 'boots');
+  ->where('name', 'John')
+  ->pull('items', 'boots'); // Pull a single value
+// Result:
+// items: ['hat', 'jeans']
+
+// Or pull multiple values
 
-$user->pull('items', 'boots');
+$user->pull('items', ['boots', 'jeans']);
+// Result:
+// items: ['hat']
 ```
 
+Also, you can pull an array value from the messages array.
```

Comment:
```suggestion
Embedded document and arrays values can also be pulled from the `messages` array.
```

Using the specific nomenclature here.

Avoiding "you" in documentation when possible is also a plus. The PHP docs are pretty good about this (see: [Style Guidelines: Personalization](http://doc.php.net/tutorial/style.php) for their reasoning).

Reply:
Thank you, Jeremy. I will keep this in mind.